lib/exportPlatform.js

```diff
@@ -93,14 +93,22 @@ function exportPlatform(platform) {
     // values like "1px solid {color.border.base}" we want to
     // transform the original value (color.border.base) before
     // replacing that value in the string.
+    const transformationsCountBefore = transformedPropRefs.length;
     const transformed = transformObject(exportableResult, platformConfig, transformationContext);
 
     // referenced values, that have not (yet) been transformed should be excluded from resolving
     const ignorePathsToResolve = deferredPropValueTransforms.map(p => getName([p, 'value']));
     exportableResult = resolveObject(transformed, {ignorePaths: ignorePathsToResolve});
-
+    console.log(transformationsCountBefore, transformedPropRefs.length, deferredPropValueTransforms.length);
     // nothing left to transform -> ready
-    if (deferredPropValueTransforms.length === 0) {
+    if (
+      // if there is no deferred value transforms left we are done
+      deferredPropValueTransforms.length === 0 ||
+      // or if we didn't transform anything since the last loop.
+      // This would happen if a token reference is not found or is circular
+      // because we keep the broken reference
+      transformedPropRefs.length === transformationsCountBefore
+    ) {
```

Comment:
I trust your logic, and it makes sense...  but it seems unintuitive to me that this is a solid "or".